src/full/Agda/TypeChecking/Rewriting.hs

```diff
@@ -352,7 +352,17 @@ checkRewriteRule q = do
     checkAxFunOrCon :: QName -> Definition -> TCM ()
     checkAxFunOrCon f def = case theDef def of
       Axiom{}        -> return ()
-      Function{}     -> return ()
+      def@Function{} -> whenJust (funProjection def) $ \proj ->
+        case projProper proj of
+          Just{} -> typeError . GenericDocError =<< hsep
+            [ prettyTCM q , " is not a legal rewrite rule, since the head symbol"
```

Comment:
Is the extra space in the string after `q` intended here?  (There is none after `f`.)